content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md

```diff
@@ -0,0 +1,40 @@
+My first move was to create a 'Simulate This Step' button which, when clicked, would produce changes in the editor corresponding to those described in the tutorial up to that point. I went digging in AutomatedTesting for scripts. Some scripts, like those to create an entity and enter game mode, were easy to simulate, but other parts were quite difficult. The 21 written steps of the tutorial also didn't each correspond to a runnable script. I identified 8 discrete parts of the tutorial, some comprising of several steps. Each discrete part could run whether or not the other parts had been run.
+
+![First conception of the simulation of steps](/images/blog/interactive-tutorials-prototyping/interactivetutorials_stepordering_1.png)
+
+Upon trying to create functions for each 'part', I soon realized that many of the parts I had identified as independent did in fact depend on each other. For example, in order to add a collider to an entity X, X must have already have been created, or else the code will break. In fact, looking at the tutorial from a higher level, there are only a few giant steps which do not depend on the previous execution of other steps.
+
+![Next conception of the simulation of steps](/images/blog/interactive-tutorials-prototyping/interactivetutorials_stepordering_1.png)
+
+This is a problem, because we don't have enough details about the state of the dice prefab in step 2 to know which sub-steps have already been simulated. We can get around this by creating giant if-statements and many near-identical state variables, but this solution is both inefficient and not applicable to other tutorials.
+
+Thus, I risked creating duplicate entities or potentially breaking the tutorial by trying to edit entities which didn't yet exist when the 'simulate' option was selection. I knew I could solve the problem by condensing the tutorial into three giant steps, but that seemed inefficient and also unhelpful for the user. I decided to create a 'simulate last step' option which displayed the completed result of the tutorial as a reference for the user.
+In the meantime, I helped move the pyside_utils.py file from EditorPythonTestTools to the QtForPython gem. The InteractiveTutorials gem uses pyside_util.py, so the move would make InteractiveTutorials usable from any project, not just AutomatedTesting. After attending a SIG-Testing meeting we updated the automated testing scripts to import pyside_utils from the new location. The experience impressed upon me the importance of communication and clarity, especially when a codebase as large as O3DE's.
```

Comment:
I would break this part into a new paragraph and mention that we had to make an RFC and get it approved, which ties in with your final sentence about the importance of communication, since we had to talk with multiple teams/stakeholders.